src/lib/Libtpp/tpp_platform.c

```diff
@@ -710,19 +710,24 @@ tpp_sock_resolve_host(char *host, int *count)
 	hints.ai_socktype = SOCK_STREAM;
 	hints.ai_protocol = IPPROTO_TCP;
 
+#ifndef WIN32
 	/* 
 	 * introducing a new mutex to prevent child process from 
 	 * inheriting getaddrinfo mutex using pthread_atfork handlers
 	 */
 	tpp_lock(&tpp_nslookup_mutex);
-
+#endif
 	if ((rc = getaddrinfo(host, NULL, &hints, &pai)) != 0) {
+#ifndef WIN32
```

Comment:
u could change the code such that there a single unlock only

rc = getaddrinfo(...);
#ifndef WIN32
      tpp_unlock(&tpp_nslookup_mutex);
#endif
     if (rc != 0) {
          error message
     }